### PR TITLE
Fix ignore on request path

### PR DESCRIPTION
**Pull request by SylvainJuge**
## What does this PR do?

Currently we ignore request using Servlet path, which does not include the Servlet context path.
As a result, trying to ignore URLs with `/app-context/*` with a web-application deployed to the `/app-context` context path is not ignored.

## Checklist

- [x] This is a bugfix
  - [x] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/master/CHANGELOG.asciidoc)
  - [x] I have added tests that would fail without this fix

### CODE REVIEW

File: CHANGELOG.asciidoc [hunk 1]
```diff
@@ -23,6 +23,18 @@ endif::[]
 [[release-notes-1.27.0]]
 ==== 1.27.0 - YYYY/MM/DD
 
+[float]
+===== Breaking changes
+* `transaction_ignore_urls` now relies on full request URL path - {pull}2146[#2146]
+** On a typical application server like Tomcat, deploying an `app.war` application to the non-ROOT context makes it accessible with `http://localhost:8080/app/`
+** Ignoring the whole webapp through `/app/*` was not possible until now.
+** We can now ignore a whole webapp through `/app/*`.
```
Comment (eyalkoren) [inline CHANGELOG.asciidoc:31]:
```suggestion
```
Seems like a duplication

File: CHANGELOG.asciidoc [hunk 2]
```diff
@@ -23,6 +23,18 @@ endif::[]
...
+** Existing configuration needs to be updated to include the deployment context, thus for example `/static/*.js` used to
```
Comment (eyalkoren) [inline CHANGELOG.asciidoc:32]:
```suggestion
** Existing configuration may need to be updated to include the deployment context, thus for example `/static/*.js` used to
```

File: CHANGELOG.asciidoc [hunk 3]
```diff
@@ -23,6 +23,18 @@ endif::[]
...
+exclude known static files in all applications might be changed to `/app/static/*.js`.
```
Comment (eyalkoren) [inline CHANGELOG.asciidoc:33]:
```suggestion
exclude known static files in all applications might be changed to `/app/static/*.js` or `*/static/*.js`.
```

File: CHANGELOG.asciidoc [hunk 4]
```diff
@@ -23,6 +23,18 @@ endif::[]
 [[release-notes-1.27.0]]
 ==== 1.27.0 - YYYY/MM/DD
 
+[float]
+===== Breaking changes
```
Comment (eyalkoren) [inline CHANGELOG.asciidoc:27]:
```suggestion
===== Potentially breaking changes
```

File: apm-agent-plugins/apm-servlet-plugin/src/test/java/co/elastic/apm/agent/servlet/helper/ServletTransactionCreationHelperTest.java [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class ServletTransactionCreationHelperTest extends AbstractInstrumentationTest {
```
Comment (eyalkoren) [inline apm-agent-plugins/apm-servlet-plugin/src/test/java/co/elastic/apm/agent/servlet/helper/ServletTransactionCreationHelperTest.java:41]: ❤️

File: apm-agent-plugins/apm-servlet-plugin/src/test/java/co/elastic/apm/agent/servlet/helper/ServletTransactionCreationHelperTest.java [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+        helper = new ServletTransactionCreationHelper(tracer);
+    }
+
+    @Test
+    void requestPathNotIgnored() {
+        HttpServletRequest request = mock(HttpServletRequest.class);
+        when(request.getRequestURI()).thenReturn("/not-ignored");
+
+        assertThat(helper.isExcluded(request)).isFalse();
+    }
```
Comment (eyalkoren) [inline apm-agent-plugins/apm-servlet-plugin/src/test/java/co/elastic/apm/agent/servlet/helper/ServletTransactionCreationHelperTest.java:58]: [Proposal] 
Extend this one a bit as well as a `@ParameterizedTest` to test:
- different configs 
- both empty and non-empty paths

**Review by eyalkoren — COMMENTED** [review]
Much better!
Some wording proposals and one test enhancement proposal, otherwise looks good!

For some reason, integration tests in the latest build took considerably longer than expected and were aborted eventually. 
I started another build, once it passes properly, you can see this PR as approved.

**Comment by SylvainJuge** [comment]
After blaming the CI, I found what was causing the timeouts.
There was a one character typo in the URL generated to poll for application status, which in turn generated invalid URLs, which could not be polled :facepalm: .

**Pull request merged**